### PR TITLE
[C++ API] L1Loss module

### PR DESCRIPTION
**Pull request by ShahriarRezghi**
@yf225 This is L1Loss module. I don't think that ```_Loss``` and ```_WeightedLoss``` as base Python classes do anything. First one sets reduction type and also takes in ```reduce``` parameter which is deprecated. The second one only registers ```weight``` parameter. I don't think that we should keep this structure. What do you think?

### CODE REVIEW

**Comment by yf225** [comment]
@ShahriarSS Not keeping `_Loss` and `_WeightedLoss` sounds good, and this can likely keep the C++ code simpler.

File: test/cpp/api/modules.cpp [hunk 1]
```diff
@@ -326,6 +327,18 @@ TEST_F(ModulesTest, Linear2_CUDA) {
...
+TEST_F(ModulesTest, L1Loss) {
+  L1Loss loss;
+  auto input = torch::randn({5,6}, torch::requires_grad());
+  auto target = torch::empty({5,6}).random_(2);
+  auto output = loss->forward(torch::sigmoid(input), target);
+  auto s = output.sum();
+  s.backward();
+
+  ASSERT_EQ(output.sizes(), torch::IntArrayRef());
+  ASSERT_EQ(input.sizes(), input.grad().sizes());
```
Comment (yf225) [inline test/cpp/api/modules.cpp]: Is there a way we can also check the correctness of `output` value after `loss->forward`?

**Comment by yf225** [comment]
I'll also work on enabling parity test for `L1Loss`, and push the changes to this PR.

File: test/cpp/api/modules.cpp (hunk 1 above)
Comment (ShahriarSS) [reply test/cpp/api/modules.cpp]: I based this on python's lL1Loss test and I think that it didn't check. What is the expected value here?

**Comment by yf225** [comment]
I will rename this PR to `[C++ API] L1Loss module` to better reflect the scope of changes.

File: test/cpp/api/modules.cpp (hunk 1 above)
Comment (yf225) [reply test/cpp/api/modules.cpp]: Got it, I think we can rely on the parity test to make sure the output value is in sync with the Python API.

**Review by yf225 — APPROVED** [review]
Thanks a lot @ShahriarSS !
